libs/wingsdk/src/ex/dynamodb-table.ts

```diff
@@ -0,0 +1,405 @@
+  /**
+   * A request to perform a update operation.
+   */
+  readonly update?: DynamodbTransactWriteItemUpdateProps;
+  /**
+   * A request to perform a delete operation.
+   */
+  readonly delete?: DynamodbTransactWriteItemDeleteProps;
+}
+
+/**
+ * Properties for `DynamodbTable.transactWriteItems`.
+ */
+export interface DynamodbTransactWriteItemsProps {
+  /**
+   * The write transact items.
+   */
+  readonly transactItems: DynamodbTransactWriteItem[];
+}
+
+/**
+ * Inflight interface for `DynamodbTable`.
+ */
+export interface IDynamodbTableClient {
+  /**
+   * Put an item into the table.
+   * @param item data to be inserted.
+   * @param props dynamodb PutItem props.
+   * @inflight
+   */
+  putItem(item: Json, props?: DynamodbTablePutItemProps): Promise<void>;
+
+  /**
+   * Get an item from the table.
+   * @param key key of the item.
+   * @param props dynamodb UpdateItem props.
+   * @inflight
+   */
+  updateItem(key: Json, props?: DynamodbTableUpdateItemProps): Promise<Json>;
+
+  /**
+   * Delete an item from the table.
+   * @param key key of the item.
+   * @inflight
+   */
+  deleteItem(key: Json): Promise<void>;
+
+  /**
+   * Get an item from the table.
+   * @param key key of the item.
+   * @inflight
+   */
+  getItem(key: Json): Promise<Json>;
+
+  /**
+   * Get the table.
+   * @inflight
+   */
+  scan(): Promise<Array<Json>>;
+
+  /**
+   * Perform a synchronous write operation that groups up to 100 action requests.
+   * @param props properties for the transact write items operation.
+   * @inflight
+   */
+  transactWriteItems(props: DynamodbTransactWriteItemsProps): Promise<void>;
+}
+
+/**
+ * List of inflight operations available for `DynamodbTable`.
+ * @internal
+ */
+export enum DynamodbTableInflightMethods {
+  /** `DynamodbTable.putItem` */
+  PUT_ITEM = "putItem",
+  /** `DynamodbTable.updateItem` */
+  UPDATE_ITEM = "updateItem",
+  /** `DynamodbTable.deleteItem` */
+  DELETE_ITEM = "deleteItem",
+  /** `DynamodbTable.getItem` */
+  GET_ITEM = "getItem",
+  /** `DynamodbTable.scan` */
+  SCAN = "scan",
+  /** `DynamodbTable.transactWriteItems` */
+  TRANSACT_WRITE_ITEMS = "transactWriteItems",
+}
+
+/**
+ * Base class for `DynamodbTable` Client.
+ */
+export abstract class DynamodbTableClientBase implements IDynamodbTableClient {
+  /**
+   * @param tableName the table name.
+   * @param attributeDefinitions the table attribute definitions.
+   * @param keySchema the table key schema.
+   */
+  constructor(
+    protected tableName: string,
+    protected attributeDefinitions: Json,
+    protected keySchema: Json
+  ) {}
+
+  /**
+   * Dynamodb table client.
+   * @internal
+   */
+  public abstract _rawClient(): Promise<DynamoDBClient>;
+
+  public async putItem(
+    item: Json,
+    props?: DynamodbTablePutItemProps
+  ): Promise<void> {
+    const client = await this._rawClient();
+    await client.send(
+      new PutItemCommand({
+        TableName: this.tableName,
+        Item: marshall(item),
```

Comment:
Does `marshall` perform validation (like telling you if there is a missing key etc.)?